MakeTimestamp/MakeTimestamp.cpp

```diff
@@ -60,9 +64,36 @@ int main(int argc, const char **argv)
 	ts.m_localHour = localST.wHour;
 	ts.m_localMinute = localST.wMinute;
 	ts.m_localSecond = localST.wSecond;
+#else
+	time_t currentTimeUnix = time(nullptr);
+
+	tm *currentTimeStruct = localtime(&currentTimeUnix);
+	if (currentTimeStruct == nullptr) {
+		fprintf(stderr, "Error converting system time to calendar format");
+		return -1;
+	}
+
+	PortabilityLayer::CombinedTimestamp ts;
+	ts.SetMacEpochTime(currentTimeUnix + ts.kMacEpochToUTC);
+
+	ts.SetLocalYear(currentTimeStruct->tm_year + 1900);
+	ts.m_localMonth = currentTimeStruct->tm_mon;
```

Comment:
kMacEpochToUTC should be subtracted from UTC time, not added.  Also, local month needs to be +1 because it is months-since-January in `struct tm`

See `GpSystemServices_POSIX::GetLocalDateTime`

Reply:
Okay, it's really stupid that `tm_mon` is 0-based but `tm_mday` is 1-based. I didn't have much of a reference for the correct format of your timestamp files, so thanks for catching this one for me.

I don't think you're right about the sign error, though. Assuming the Mac timestamp counts the number of seconds since `1904-01-01 00:00 UTC`, which is the only sane interpretation I can think of,
```
2024-08-06 09:00 UTC
= 1722934800 seconds since 1970-01-01 00:00 UTC
= 3805779600 seconds since 1904-01-01 00:00 UTC

(1970-01-01 00:00 UTC) - (1904-01-01 00:00 UTC) = 2082844800
3805779600 = 1722934800 + 2082844800
```

Subtracting 2082844800 as per `GpSystemServices_POSIX::GetTime()`:
```
1722934800 - 2082844800 = -359910000

(1904-01-01 00:00 UTC) + (-359910000) = 1892-08-04 09:00 UTC
```

That clearly isn't right, so it must be the other way.

Maybe `kMacEpochToUTC` isn't the best name for a magic number of `2082844800` but either it needs to stay positive and every instance of converting Unix → Mac should **add** this constant, and **subtract** for the reverse conversion, or the sign of `kMacEpochToUTC` needs to be flipped and every instance of converting Unix → Mac should **subtract** it, and **add** for the reverse.

If you intended the number to be the Mac epoch represented as a Unix timestamp, it should be a negative number.